docs/modules/ROOT/pages/explanations/decisions/api-design.adoc

```diff
@@ -0,0 +1,243 @@
+= Decision on API Design
+
+== Problem
+
+This decision is about how we can achieve a "common look" as defined https://kb.vshn.ch/app-catalog/explanations/why_exists.html#_standard_way_to_provide_services[here].
+
+The ideal way would be, that we can re-use the exact same XRD for each set of services.
+For example all implementations for PostgreSQL have exactly the same API for the user.
+Due to the nature of the product, not all implementations provide the same functionality.
+The set of available features is most likely not the same between most providers.
+Also with Crossplane we're a bit limited on what's possible with translating the source API spec into the target API spec.
+
+== Requirements
+
+* APIs for a set of similar services should feel similar
+* They should only need minimal configuration for a default case
+* We should be able to leverage unique features of the service providers
+
+== Proposals
+
+=== Proposal 1: Use same XRD with provider specific fields
+
+There's exactly one XRD for a set of services.
+This XRD contains a minimal set of common parameters, as well as a field for provider specific configurations.
+
+.Example Claim
+[source, yaml]
+----
+apiVersion: appcat.vshn.io/v1
+kind: PostgreSQL
+metadata:
+  name: my-instance
+spec:
+  parameters:
+    name: my-instance
+    majorVersion: '14'
+    pgSettings: {}
+    exoscale:
+      plan: hobbyist-2
+      type: pg
+      maintenance:
+        day: 2
+        hour: 22
+        minute: 0
+      backup:
+        hour: 20
+        minute: 0
+    vshn:
+      minorVersion: '4'
+----
+
+Advantages::
+
+* We can use the same XRD
+* Adding new providers to the list is easy
+
+Disadvantages::
+
+* We don't really have a "common look"
+* The customer needs to know what the underlying service provider is, even for simple instances
+
+=== Proposal 2: Separate XRDs
```

Comment:
Maybe use the same semantics as Crossplane does? At the top-level (`spec.X`) _always_ have the same parameters, and then have a provider specific part like `spec.forProvider.X`.

Reply:
I'm not quite sure what you mean here. Both examples have the same parameters under `spec.parameters.*`?

Reply:
My concern is, that `.spec.size` is has different options, depending on the provider used, which might be confusing. E.g. on one we have `.spec.size.plan` and on another we have `.spec.size.cpu`. So my proposal (maybe a new one) would be:

```
apiVersion: exoscale.appcat.vshn.io/v1
kind: PostgreSQL
metadata:
  name: my-instance
spec:
  parameters:
    name: my-instance
    version: '14'
    maintenance:
      day: 2
      hour: 22
      minute: 0
    backup:
      hour: 20
      minute: 0
    providerSpecific:
      service:
        type: pg
        settings: {}
        plan: hobbyist-2
```

This introduces the `providerSpecific` field, which is different for each provider, while the other are always the same.

Reply:
Added it as another proposal.